### PR TITLE
Improve laxness of certificate parsing

### PR DESCRIPTION
**Pull request by daviddrysdale**
Make lax ASN.1 parsing explicitly controlled, and turn it on for cert parsing of key fields (and emit non-fatal errors when lax parsing succeeds but strict parsing fails).

Along the way, relax:
 - checking of integer minimal encoding
 - checking of non-negative RSA modulus.

### CODE REVIEW

**Comment by daviddrysdale** [comment]
Might be easiest reviewed commit-by-commit

File: asn1/asn1.go [hunk 1]
```diff
@@ -9,9 +9,19 @@
...
-// for errors (e.g. use of tagPrintableString when the string data is really
-// ISO8859-1 - a common error present in many x509 certificates in the wild.)
+// (encoding/asn1), with the aim of relaxing checks for various things
+// that are common errors present in many X.509 certificates in the
+// wild.
+//
+// Main differences:
+//  - Extra "lax" tag that recursively applies and relaxes some strict
+//    checks:
+//     - parsePrintableString() copes with unvalid PrintableString contents,
```
Comment (AlCutter) [inline asn1/asn1.go]: unvalid->invalid

File: asn1/common.go [hunk 2]
```diff
@@ -81,6 +81,7 @@ type fieldParameters struct {
 	timeType     int    // the time tag to use when marshaling.
 	set          bool   // true iff this should be encoded as a SET
 	omitEmpty    bool   // true iff this should be omitted if empty when marshaling.
+	lax          bool   // true iff unmarshalling should skip some steps
```
Comment (AlCutter) [inline asn1/common.go]: some steps -> some verification steps?

File: asn1/asn1.go (hunk 1 above)
Comment (daviddrysdale) [reply asn1/asn1.go]: Done.

File: asn1/common.go (hunk 2 above)
Comment (daviddrysdale) [reply asn1/common.go]: Done.

**Pull request merged**